tests/conftest.py

```diff
@@ -48,39 +80,252 @@ def swg_config(indexd_client): # noqa
+
+
+@pytest.fixture
+def swg_bulk_client_no_migrate(swg_config):
+    return swagger_client.BulkApi(swagger_client.ApiClient(swg_config_no_migrate))
+
+
+@pytest.fixture
+def index_driver():
+    driver = SQLAlchemyIndexDriver(PG_URL)
+    yield driver
+    drop_tables(driver, index_base)
+    driver.dispose()
+
+
+@pytest.fixture
+def alias_driver():
+    driver = SQLAlchemyAliasDriver(PG_URL)
+    yield driver
+    drop_tables(driver, alias_base)
+    driver.dispose()
+
+
+@pytest.fixture
+def auth_driver():
+    driver = SQLAlchemyAuthDriver(PG_URL)
+    yield driver
+    drop_tables(driver, auth_base)
+    driver.dispose()
+
+
+@pytest.fixture
+def index_driver_no_migrate():
+    """
+    This fixture is designed for testing migration scripts and can be used for
+    any other situation where a migration is not desired on instantiation.
+    """
+    driver = SQLAlchemyIndexDriver(PG_URL, auto_migrate=False)
+    yield driver
+    drop_tables(driver, index_base)
+    driver.dispose()
+
+
+@pytest.fixture
+def alias_driver_no_migrate():
+    """
+    This fixture is designed for testing migration scripts and can be used for
+    any other situation where a migration is not desired on instantiation.
+    """
+    driver = SQLAlchemyAliasDriver(PG_URL, auto_migrate=False)
+    yield driver
+    drop_tables(driver, alias_base)
+    driver.dispose()
+
+
+@pytest.fixture
+def create_tables(index_driver, alias_driver, auth_driver):
+    """Make sure the tables are created but don't operate on them directly.
+
+    Also set up the password to be accessed by the client tests.
+    """
+    auth_driver.add('admin', 'admin')
+
+
+@pytest.fixture
+def create_tables_no_migrate(
+        index_driver_no_migrate, alias_driver_no_migrate, auth_driver):
+    """Make sure the tables are created but don't operate on them directly.
+
+    There is no migration required for the SQLAlchemyAuthDriver.
+    Also set up the password to be accessed by the client tests.
+    """
+    auth_driver.add('admin', 'admin')
+
+
+@pytest.fixture
+def database_engine():
+    engine = create_engine(PG_URL)
+    yield engine
+    engine.dispose()
+
+
+@pytest.fixture
+def database_conn(database_engine):
+    conn = database_engine.connect()
+    yield conn
+    conn.close()
+
+
+def drop_tables(driver, base):
+    """Drop all the tables in this application's scope.
+
+    This has the same effect as deleting the sqlite file. Your test will have a
+    fresh database for it's run.
+    """
+
+    with driver.session:
+        # Drop tables in reverse order to avoid cascade drop errors.
+        # metadata is a sqlalchemy property.
+        # sorted_tables is a list of tables sorted by their dependencies.
+        for table in reversed(base.metadata.sorted_tables):
+            # Check first to see if the table exists before dropping it.
+            table.drop(checkfirst=True)
+
+
+@pytest.fixture(scope='session')
+def indexd_server():
```

Comment:
take a look at live_server from pytest-flask. Feels like it duplicating that but also is functionally scoped so I get it.

We should be able to run our tests in both debug and not debug mode with flask.

https://github.com/pytest-dev/pytest-flask/blob/master/pytest_flask/fixtures.py#L125

Which of course can be ingested with another pytest-plugin
https://pypi.org/project/pytest-base-url/